common/lib/xmodule/xmodule/js/src/problem/edit.coffee

```diff
@@ -42,8 +48,19 @@ class @MarkdownEditingDescriptor extends XModule.Descriptor
     lineNumbers: true
     lineWrapping: true
     })
+
     if text
       @xml_editor.setValue(text)
+
+    totalLines = @xml_editor.lineCount()
+    totalChars = @xml_editor.getTextArea().value.length
+    @xml_editor.autoFormatRange {
```

Comment:
yeah we are using version 3.15 in the application. I am now implementing a separate logic with the help of a library to format xml before handing this over to code mirror.